firedrake/function.py

```diff
@@ -354,13 +360,22 @@ def _interpolate(self, fs, dat, expression, subset):
             raise RuntimeError('Rank mismatch: Expression rank %d, FunctionSpace rank %d'
```

Comment:
The `expression.rank` call now raise a UFL warning when interpolating a UFL expression:

```python
 f.interpolate(as_tensor([x[0],x[1]]))
Expr.rank() is deprecated, please use len(expr.ufl_shape) instead.
```

Can you change to the new syntax?